lib/web_ui/test/text_editing_test.dart

```diff
@@ -390,6 +390,28 @@ Future<void> testMain() async {
       expect(event.defaultPrevented, isFalse);
     });
 
+    test('Triggers input action and prevent new line key event for single line field', () {
```

Comment:
This looks great! 

@mdebbar would it be worth adding a test to verify return key behavior for multiline inputs is preserved?

Reply:
There is already a test for that right above this :)

Reply:
@htoor3 If I remember well this is already covered by an existing test. (I first I implemented this change for both single line and multiline inputs and it broke several tests. I ended targeting only single line, not because of those failing tests, but because changing the behavior for multilines inputs will require some discussions. I will open a new issue to discuss this, see https://github.com/flutter/flutter/issues/113559#issuecomment-1285665506).